test_autocti/extract/two_d/serial/test_serial_fpr.py

```diff
@@ -74,6 +74,51 @@ def test__region_list_from__array_2d_list_from(serial_array, serial_masked_array
     ).all()
 
 
+def test__region_list_from__array_2d_list_from__pixels_from_end(
+    serial_array, serial_masked_array
+):
+    extract = ac.Extract2DSerialFPR(
+        shape_2d=serial_array.shape_native, region_list=[(0, 3, 1, 4)]
+    )
+
+    fpr = extract.array_2d_list_from(array=serial_array, pixels_from_end=1)
+
+    assert (fpr == np.array([[3.0], [3.0], [3.0]])).all()
+
+    extract = ac.Extract2DSerialFPR(
+        shape_2d=serial_array.shape_native, region_list=[(0, 3, 1, 4), (0, 3, 5, 8)]
+    )
+
+    fpr_list = extract.array_2d_list_from(array=serial_array, pixels_from_end=1)
+
+    assert (fpr_list[0] == np.array([[3.0], [3.0], [3.0]])).all()
+    assert (fpr_list[1] == np.array([[7.0], [7.0], [7.0]])).all()
+
+    fpr_list = extract.array_2d_list_from(array=serial_array, pixels_from_end=3)
+
+    assert (
+        fpr_list[0] == np.array([[1.0, 2.0, 3.0], [1.0, 2.0, 3.0], [1.0, 2.0, 3.0]])
+    ).all()
+
+    assert (
+        fpr_list[1] == np.array([[5.0, 6.0, 7.0], [5.0, 6.0, 7.0], [5.0, 6.0, 7.0]])
+    ).all()
+
+    fpr_list = extract.array_2d_list_from(array=serial_masked_array, pixels_from_end=3)
+
+    assert (
+        (fpr_list[0].mask)
+        == np.array(
+            [[False, False, False], [False, True, False], [False, False, False]]
+        )
+    ).all()
+
+    assert (
+        fpr_list[1].mask
+        == np.array([[True, False, False], [False, True, False], [False, False, True]])

```

Comment:
I think if you put a comma before the closing bracket this will format over new lines which I find clearer for 2D arrays